pkg/eventhandler/drainer/drainer.go

```diff
@@ -33,35 +34,57 @@ func NewDrainEventHandler(logger micrologger.Logger, client *azuremetadataclient
 
 func (s *DrainEventHandler) HandleEvent(ctx context.Context, event azuremetadataclient.ScheduledEvent) error {
 	s.Logger.Debugf(ctx, "Received event: %v", event)
-	if event.EventType == "Terminate" && event.ResourceType == "VirtualMachine" {
-		s.Logger.LogCtx(ctx, "message", "found Terminate event, start draining the node")
+
+	var timeout time.Duration
+	{
+		t, err := time.Parse(time.RFC1123, event.NotBefore)
+		if err != nil {
+			timeout = 15 * time.Minute
+		} else {
+			timeout = time.Until(t)
+		}
+	}
+
+	switch event.EventType {
+	case "Terminate":
+	case "Preempt":
+	case "Reboot":
+	case "Redeploy":
+	default:
+		s.Logger.Debugf(ctx, "Warning: unhandled event type: %s (resource type %s)", event.EventType, event.ResourceType)
+		return nil
+	}
+
+	// If timeout is negative we have no time to drain the node, so we skip the step.
+	if timeout > 0 {
+		s.Logger.Debugf(ctx, "got event %s, start draining the node (timeout %.0f seconds)", event.EventType, timeout.Seconds())
 
 		// Drain the node.
-		err := s.drainNode(ctx, s.K8sClient, s.LocalNodeName)
+		err := s.drainNode(ctx, s.K8sClient, s.LocalNodeName, timeout)
 		if IsEvictionInProgress(err) {
 			s.Logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("node %q not drained in time.", s.LocalNodeName))
 		} else if err != nil {
 			return microerror.Mask(err)
 		} else {
 			s.Logger.Debugf(ctx, fmt.Sprintf("node %q drained successfully.", s.LocalNodeName))
 		}
+	}
 
-		// Delete node from k8s.
-		s.Logger.Debugf(ctx, "Deleting Node %q from k8s API", s.LocalNodeName)
-		err = s.K8sClient.CoreV1().Nodes().Delete(ctx, s.LocalNodeName, v1.DeleteOptions{})
-		if apierrors.IsNotFound(err) {
-			s.Logger.Debugf(ctx, "Node %q was not found, it was probably already deleted", s.LocalNodeName)
-		} else if err != nil {
-			s.Logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("Error deleting node %q from Kubernetes API: %s.", s.LocalNodeName, err))
-		}
+	// Delete node from k8s.
+	s.Logger.Debugf(ctx, "Deleting Node %q from k8s API", s.LocalNodeName)
+	err := s.K8sClient.CoreV1().Nodes().Delete(ctx, s.LocalNodeName, v1.DeleteOptions{})
```

Comment:
It's probably ok, but just saying: Now node reboot makes it look new :smile:

Reply:
Yes, true. The other option would have been to find a way to uncordon the node after reboot and that was dangerous imho

Reply:
Sure. It could be even useful to see "uptime".